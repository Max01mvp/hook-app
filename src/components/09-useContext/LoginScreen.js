import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  
  //1. Obtener la refencia al userContext

    const { setUser } = useContext( UserContext );

    //2. setUser

    /*
        id:123,
        name: 'John Doe',
    */

  return (
    <div>
        <h1>Login Screen</h1>
        <hr />
        <button 
        
        className="btn btn-primary"
        onClick={ () => setUser({

          id: 123,
          name: 'John Doe',

        })}
        >
          Login
        </button>
        
    </div>
  )
}
