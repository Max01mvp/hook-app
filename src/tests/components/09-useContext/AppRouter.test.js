import React from 'react'
import { shallow } from 'enzyme'
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';
describe('Pruebas en <AppRouter />', () => {


    const user = {

        id: 1,
        name: 'Max'

    }


    const wrapper = shallow( 
        <UserContext.Provider value={{
            user
        }}>
                <AppRouter /> 
        </UserContext.Provider>
    );

    test('Debe de mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    })


})