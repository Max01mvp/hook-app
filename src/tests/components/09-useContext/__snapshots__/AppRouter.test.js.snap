// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrarse correctamente 1`] = `<AppRouter />`;
